### DISCUSSION TITLE
Pushing to Github using Personal Access Token not working

### DISCUSSION THREAD

**Discussion by guganathan1988** [community/community#63672] [New to GitHub]
I getting error on my console when I tried to use git push, gitup not access token authentication

Error as ; remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/guganathan1988/firstrepo224.git/

**Comment by nirav-gajera** [comment]
you need to use a personal access token (PAT) to authenticate with GitHub. A PAT is a secure, reusable token that you can use to access GitHub without having to enter your password every time.

To create a PAT, you can follow these steps:

1. Go to GitHub and log in to your account.
2. Click on your profile picture in the top right corner of the page and select "Settings".
3. Click on the "Developer settings" tab.
4. Click on the "Personal access tokens" section.
5. Click on the "Generate new token" button.
6. Give your token a name and select the scopes that you want it to have. The "repo" scope is the only scope that you need to push to GitHub.
7. Click on the "Generate token" button.

Once you have created your PAT, you can use it to authenticate with GitHub by setting the `GITHUB_TOKEN` environment variable. You can do this by running the following command in your terminal:

```
export GITHUB_TOKEN=<your_pat>
```

You can then use git push to push your changes to GitHub. For example, to push your changes to the master branch, you would run the following command:

```
git push origin master
```

This will use your PAT to authenticate with GitHub and push your changes to the master branch.

**Reply by guganathan1988** [reply]
export GITHUB_TOKEN=<my token>, it showing incorrect syntax in my terminal sir...

**Reply by nirav-gajera** [reply]
Replaced `YOUR_PAT` with the value of your PAT(personal access token) ??

**Comment by Vladmaster99** [comment]
Github Personal access token not working in Windows 7 have tried .gitconfig  and crediatial manager and added to dialog box for password. Also tried set Github_Token= ghp_blahblahblah. But all fail. Tried also on Windows XP but this fails too.,

**Comment by dariusz-bzowka-chain-insight** [comment]
is it so hard to provide an example url?
